### PR TITLE
week 7 homework

### PR DESCRIPTION
**Pull request by cherimarie**

### CODE REVIEW

File: week7/homework/questions.txt [hunk 1]
```diff
@@ -3,7 +3,27 @@ Please Read Chapters 23 and 24 DuckTyping and MetaProgramming
...
 1. What is method_missing and how can it be used?
-2. What is and Eigenclass and what is it used for? Where Do Singleton methods live?
+*It's the method that Ruby looks for after it can't find the method that's been called anywhere in the class heirarchy that it was called in.
+
+
+2. What is and Eigenclass and what is it used for? 
+*The singleton class, or Eigneclass, is an anonymous class that bursts into appearance when a singleton method is is defined. It holds the definition of the singleton method. It becomes the class of the object that the singleton method is attached to, pushing that object's original class up a level to become its superclass.
+
+2.5. Where Do Singleton methods live?
+*In yurts.
```
Comment (nellshamrell) [inline week7/homework/questions.txt:13]: LOL!!!!!

Comment (nellshamrell) [inline week7/homework/questions.txt:13]: Kelsey just came running in "What???"

File: week7/homework/questions.txt [hunk 2]
```diff
@@ -3,7 +3,27 @@ Please Read Chapters 23 and 24 DuckTyping and MetaProgramming
...
+*A class method is a method that's available to a class. It also doesn't exist or is identical to a singleton method that's called on the self object that is the class. 
+*Instance methods can be called from inside (or sometimes outside) their class. They have access to the object's instance variables and state.
+
+4.5. What is the difference between instance_eval and class_eval?
+*They both allow you to change the identity of 'self' for the duration of a block. class_eval sets things up as if you were in the body of a class definition. instance_eval acts as though you were wroking in side the singleton class of 'self'.
+
 5. What is the difference between a singleton class and a singleton method?
+*A singleton method is a method that is specific to a particular object (as opposed to a class).
+*(see question 2)The singleton class, or Eigneclass, is an anonymous class that bursts into appearance when a singleton method is is defined. It holds the definition of the singleton method. It becomes the class of the object that the singleton method is attached to, pushing that object's original class up a level to become its superclass.
```
Comment (nellshamrell) [inline week7/homework/questions.txt:29]: I like that "bursts into appearance"

File: week7/homework/questions.txt (hunk 1 above)
Comment (cherimarie) [inline week7/homework/questions.txt:13]: Just be all "Yurts!" She'll totally get it. Yurts are universally hilarious
:)

File: week7/homework/questions.txt (hunk 2 above)
Comment (cherimarie) [inline week7/homework/questions.txt:29]: Creative writing makes homework more interesting. And keeps me from
accidentally copying verbatim from the book. I'm glad it's making it more
fun for you, too!

**Comment by nellshamrell** [comment]
Homework looks good!

**Comment by cherimarie** [comment]
Thanks. I felt like I did well on Week7. That hellish midterm experience
was definitely a turning point for me :)

File: week7/homework/questions.txt (hunk 1 above)
Comment (nellshamrell) [inline week7/homework/questions.txt:13]: We were discussing them earlier this evening, that's part of what made it so hilarious :)

File: week7/homework/questions.txt (hunk 2 above)
Comment (nellshamrell) [inline week7/homework/questions.txt:29]: Agreed!

**Pull request closed without merging**